kie-internal/src/main/java/org/kie/internal/services/KieAssemblersImpl.java

```diff
@@ -39,6 +39,19 @@ public KieAssemblersImpl() {
         return assemblers;
     }
 
+    @Override
+    public void addResourceAsPackageDescr(Object knowledgeBuilder, Resource resource, ResourceType type, ResourceConfiguration configuration) throws Exception {
```

Comment:
@evacchi 
Sorry, I can't get that. There is a method 
`addResourceAsPackageDescr` whose name clearly suggest the added resource is a `PackageDescr` (and only that).
But this method invoke `addResourceBeforeRules` whose name (and semantic) is different, i.e. define to add a resource (regardless of generated type) before rules are compiled. 
Just to be overtly pedantic, the former method has a "what" semantic, while the latter has a "when" semantic.
I think this is pretty misleading. Beside that, I think that the "what"  should be an implementation detail, and that all those methods should only refer to the "when".
Long story short: could you please rename  `addResourceAsPackageDescr` to `addResourceBeforeRules` ? This would make it consistent and clear everywhere it is read/used

Reply:
yes. I actually renamed the other one too and only forgot this one :P

Reply:
(see previous commit :P)

Reply:
:+1: